src/Components/AuthFormFunction.js

```diff
@@ -0,0 +1,63 @@
+import { auth } from "../firebase";
+import {
+  signInWithEmailAndPassword,
+  createUserWithEmailAndPassword,
+} from "firebase/auth";
+import { useState } from "react";
+import { useNavigate, Link } from "react-router-dom";
+
+export const AuthFormFunction = () => {
+  const navigate = useNavigate();
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  };
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+  };
```

Comment:
Do we need functions for the state updates? I think you can just make the state update in the onChange itself